tests/phpunit/unit/Extension/Mock/TwigExtension.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Bolt\Tests\Extension\Mock;
+
+use Bolt\Extension\SimpleExtension;
+
+/**
+ * Mock extension that extends SimpleExtension for testing the TwigTrait.
+ *
+ * @author Gawain Lynch <gawain.lynch@gmail.com>
+ */
+class TwigExtension extends SimpleExtension
+{
+    public function isSafe()
+    {
+        return true;
+    }
+
+    public function getTestTemplateOutput($template, array $context = [])
+    {
+        return $this->renderTemplate($template, $context);
+    }
+
+    public function twigFunctionCallback($input)
+    {
+        return strtolower($input);
+    }
+
+    public function twigFilterCallback($input)
+    {
+        return strtoupper($input);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function registerTwigFunctions()
+    {
+        return [
+            'koala'    => 'twigFunctionCallback',
+            'dropbear' => ['twigFunctionCallback', ['safe' => true]],
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function registerTwigFilters()
+    {
+        return [
+            'koala'    => 'twigFilterCallback',
+            'dropbear' => ['twigFilterCallback', ['safe' => true]],
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function registerTwigPaths()
+    {
+        return [
+            'koala',
+            'dropbear' => ['position' => 'prepend', 'namespace' => 'Marsupial'],
```

Comment:
Sure, but it is more than we have, and I have very little desire/allowance time to work on tests and am trying to turn a few quick PRs over to move a few things along … with the hope it makes it easier for *others* to help.